drivers/sdcard_spi/sdcard_spi.c

```diff
@@ -134,49 +217,43 @@ static sd_init_fsm_state_t _init_sd_fsm_step(sdcard_spi_t *card, sd_init_fsm_sta
             ztimer_sleep(ZTIMER_USEC, SD_CARD_WAIT_AFTER_POWER_UP_US);
         }
 
-        gpio_set(card->params.mosi);
-        gpio_set(card->params.cs);       /* unselect sdcard for power up sequence */
-
-        /* powersequence: perform at least 74 clockcycles with mosi_pin being high
-         * (same as sending dummy bytes with 0xFF) */
+        /* powersequence: perform at least 74 clockcycles sending dummy bytes with 0xFF,
+           (same as same clock cycles with mosi_pin being high) */
+        _select_card_spi(card, SPI_CLK_100KHZ, IS_USED(MODULE_PERIPH_SPI_GPIO_MODE));
+        /* dont select card for power up sequence */
+        gpio_set(card->params.cs);
+        uint8_t dummy;
         for (int i = 0; i < SD_POWERSEQUENCE_CLOCK_COUNT; i += 1) {
-            gpio_set(card->params.clk);
-            ztimer_sleep(ZTIMER_USEC, SD_CARD_PREINIT_CLOCK_PERIOD_US / 2);
-            gpio_clear(card->params.clk);
-            ztimer_sleep(ZTIMER_USEC, SD_CARD_PREINIT_CLOCK_PERIOD_US / 2);
+            _dyn_spi_rxtx_byte(card, SD_CARD_DUMMY_BYTE, &dummy);
         }
```

Comment:
this used to send `SD_POWERSEQUENCE_CLOCK_COUNT` bits, now it sends `SD_POWERSEQUENCE_CLOCK_COUNT` bytes.